File: 0517-Super Washing Machines/0517-Super Washing Machines.cpp
class Solution {
public:
    int findMinMoves(vector<int>& machines) {
        int n = machines.size();
        int sum = accumulate(machines.begin(), machines.end(), 0);
        if (sum % n)
        {
            return -1;
        }
        
        int avg = sum / n;
        int count = 0;
        int curr = 0;
        for (int i = 0; i < n; ++i)
        {
            int diff = machines[i] - avg;
            curr += diff;
            count = max(count, max(abs(curr), diff));
        }
        
        return count;
    }
};

File: 0501-0600/0517-Super Washing Machines/0517-Super Washing Machines.py
class Solution:
    def findMinMoves(self, machines: List[int]) -> int:
        total = sum(machines)
        if total % len(machines):
            return -1
        avg = total // len(machines)
        curr = count = 0 
        for m in machines:
            diff = m - avg
            curr += diff
            count = max(count, abs(curr), diff)
        return count
